packages/trpc/server/routers/loggedInViewer/integrations.handler.ts

```diff
@@ -144,56 +52,21 @@ export const integrationsHandler = async ({ ctx, input }: IntegrationsOptions) =
       const invalidCredentialIds = credentials
         .filter((c) => c.appId === app.slug && c.invalid)
         .map((c) => c.id);
-      const teams = await Promise.all(
-        credentials
-          .filter((c) => c.appId === app.slug && c.teamId)
-          .map(async (c) => {
-            const team = userTeams.find((team) => team.id === c.teamId);
-            if (!team) {
-              return null;
-            }
-            return {
-              teamId: team.id,
-              name: team.name,
-              logoUrl: team.logoUrl,
-              credentialId: c.id,
-              isAdmin:
-                team.members[0].role === MembershipRole.ADMIN ||
-                team.members[0].role === MembershipRole.OWNER,
-            };
-          })
+      const teams = await constructUserTeams(credentials, app.slug, userTeams);
+      const dependencyData = getAppDependencyData(enabledApps, app.dependencies);
+      const isSetupAlready = await checkAppSetupStatus(
+        credential,
+        app.categories.includes("payment"),
+        app.dirName
       );
+
       // type infer as CredentialOwner
       const credentialOwner: CredentialOwner = {
         name: user.name,
         avatar: user.avatar,
       };
 
-      // We need to know if app is payment type
-      // undefined it means that app don't require app/setup/page
-      let isSetupAlready = undefined;
-      if (credential && app.categories.includes("payment")) {
-        const paymentApp = (await appStore[app.dirName as keyof typeof appStore]?.()) as PaymentApp | null;
-        if (paymentApp && "lib" in paymentApp && paymentApp?.lib && "PaymentService" in paymentApp?.lib) {
-          const PaymentService = paymentApp.lib.PaymentService;
-          const paymentInstance = new PaymentService(credential);
-          isSetupAlready = paymentInstance.isSetupAlready();
-        }
-      }
-
-      let dependencyData: TDependencyData = [];
-      if (app.dependencies?.length) {
-        dependencyData = app.dependencies.map((dependency) => {
-          const dependencyInstalled = enabledApps.some(
-            (dbAppIterator) => dbAppIterator.credentials.length && dbAppIterator.slug === dependency
-          );
-          // If the app marked as dependency is simply deleted from the codebase, we can have the situation where App is marked installed in DB but we couldn't get the app.
-          const dependencyName = getAppFromSlug(dependency)?.name;
-          return { name: dependencyName, installed: dependencyInstalled };
-        });
-      }
-
-      return {
+      const appData = {
```

Comment:
```suggestion
      return {
```

Again. The constant is not being used anywhere else.